WcaOnRails/app/models/team_member.rb

```diff
@@ -24,6 +44,34 @@ def cannot_demote_oneself
     end
   end
 
+  validate :delegate_team_must_have_at_least_one_senior_delegate
+  def delegate_team_must_have_at_least_one_senior_delegate
+    if self.team.committee.slug == "wca-delegates-committee"
+      unless self.committee_position.slug == 'senior-delegate' && self.current_member?
```

Comment:
HI @jonatanklosko, unfortunately this didn't work :( 

The problem is that self.user.senior_delegate? expects the record to already be saved in the database. As we are validating the team_member this doesn't work. I'll switch it to an if statement.